src/Squirrel/ReleasePackage.cs

```diff
@@ -197,50 +187,67 @@ public static Task ExtractZipForInstall(string zipFilePath, string outFolder, st
 
         public static Task ExtractZipForInstall(string zipFilePath, string outFolder, string rootPackageFolder, Action<int> progress)
         {
-            var re = new Regex(@"lib[\\\/][^\\\/]*[\\\/]", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
+			// Regex for the matching of "lib/[framework]/" directory.
+			var re = new Regex(@"lib[\\\/][^\\\/]*[\\\/]", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
 
             return Task.Run(() => {
-                using (var za = ZipArchive.Open(zipFilePath))
-                using (var reader = za.ExtractAllEntries()) {
-                    var totalItems = za.Entries.Count;
+				using (ZipFile zip = ZipFile.Read(zipFilePath)) {
+					var totalItems = zip.Entries.Count;
                     var currentItem = 0;
 
-                    while (reader.MoveToNextEntry()) {
-                        // Report progress early since we might be need to continue for non-matches
-                        currentItem++;
-                        var percentage = (currentItem * 100d) / totalItems;
-                        progress((int)percentage);
+					foreach (ZipEntry entry in zip.Entries.ToList()) {
+						string folderToExtractEntry = outFolder;
 
-                        var parts = reader.Entry.Key.Split('\\', '/');
-                        var decoded = String.Join(Path.DirectorySeparatorChar.ToString(), parts);
+						// Report progress early since we might be need to continue for non-matches
+						currentItem++;
+						var percentage = (currentItem * 100d) / totalItems;
+						progress((int)percentage);						
 
-                        if (!re.IsMatch(decoded)) continue;
-                        decoded = re.Replace(decoded, "", 1);
+						// Skip the extracting of entries which path does not contain "lib/[framework]/". 
+						if (!re.IsMatch(entry.FileName)) continue;
 
-                        var fullTargetFile = Path.Combine(outFolder, decoded);
-                        var fullTargetDir = Path.GetDirectoryName(fullTargetFile);
-                        Directory.CreateDirectory(fullTargetDir);
+						// Cut "lib/[framework]/" from the entry FileName.
+						var cutPath = re.Replace(entry.FileName, "", 1);
+
+						// Skip the extracting of "lib/[framework]/" directory.
+						if (cutPath == "") continue;						
 
                         var failureIsOkay = false;
```

Comment:
What does this variable mean? Can you add a short comment?

Reply:
Yeah we should refactor it and add comments in the future